### PR TITLE
Build ES5/ES6 in parallel to share gulp deps

### PR DESCRIPTION
**Pull request by emlove**
Currently, build tasks are getting duplicated in the build output. (Notably build-translations).

By building ES5/ES6 in parallel, vs. in a run sequence, we can let gulp run the dependent tasks only once.

### CODE REVIEW

**Review by andrey-git — CHANGES_REQUESTED** [review]
The two versions of roll-up both use build-temp so can't run in parallel.

**Comment by andrey-git** [comment]
Note that since automation.js and script-editor.js are included from `build-temp` it is no longer possible to develop with those files transpiled. I guess that's ok...

**Review by andrey-git — COMMENTED** [review]
Also fix path of compatibility in index.html and gulp/tasks/gen-index-html.js

**Comment by armills** [comment]
Hah, still needed to rebase on the latest.

Let me think about the included JS files for a bit.

**Comment by balloob** [comment]
It's fine to not be able to develop with transpiled versions as we are not able to do that anyway, since we point at an untranspiled entrypoint.

Can the files in `build-temp` just be untranspiled and the ES5 build step will transpile it when it is including it in the bundle?

**Comment by andrey-git** [comment]
automation and script JS files are React, so they are still transpiled to "normal" JS even when not using es5

File: gulp/tasks/hassio-panel.js [hunk 1]
```diff
@@ -37,8 +37,8 @@ async function buildHassioPanel(es6) {
   });
 
   return minifyStream(stream, es6)
-    .pipe(rename(`hassio-main-${es6 ? 'latest' : 'es5'}.html`))
-    .pipe(gulp.dest('build-temp'));
+    .pipe(rename('hassio-main.html'))
+    .pipe(gulp.dest(es6 ? 'build-temp' : 'build-temp-es5'));
```
Comment (armills) [inline gulp/tasks/hassio-panel.js]: I believe this branch is mostly good to go, but I'm not really familiar with the hassio build process. I couldn't find where this is used. I'm not a hassio user myself.

Comment (balloob) [reply gulp/tasks/hassio-panel.js]: https://github.com/home-assistant/hassio-build/tree/master/build-scripts/frontend-build-env and then with this PR added https://github.com/home-assistant/hassio-build/pull/47

Comment (balloob) [reply gulp/tasks/hassio-panel.js]: So if we change this, I should update that PR

**Review by balloob — APPROVED** [review]

**Comment by andrey-git** [comment]
I think this introduces a race condition and possibly breaks the build script. Will check later :)

**Comment by balloob** [comment]
What's there to race if we always have independent folders for all artifacts?

**Comment by andrey-git** [comment]
It is caused by html files having `<script src='../../../build-temp/script-editor.js'>` imports.

I think this causes 2 issues:
1) Very unlikely but theoretically possible that by the time polymer-bundle in build_es5 reaches those files they are not built yet by rollup_es6.

2) Even if they are built, will build_es5 transpile them to es5? If yes, there is no need to transpile them and put them in build-temp-es5 by rollup_es5 in the first place.

**Comment by balloob** [comment]
Oh, good point. Doh.

I just confirmed, any JavaScript referenced from Polymer does not need to be pre-transpiled.
 We don't need to generate any ES5 javascript except for the ones that we move to `build-es5`: `core.js` and `compatibility.js`

**Comment by balloob** [comment]
I hope BTW that with Polymer 3 we can get rid of a lot of this custom build chain stuff. It's getting messy.

**Comment by armills** [comment]
@andrey-git I think you're right, but the fix should just be a matter of including the es6 rollup as a dependency for the polymer bundle es5 build. I thought I had done that, but it looks like I missed it here.

And I agree, we should be able to drop everything but `core.js` and `compatibility.js` from the es5 rollup.

**Pull request merged**